plugins/common/ahutil/ahutil.go

```diff
@@ -0,0 +1,358 @@
+//#endif
+//#ifdef AH_SUPPORT_RX_SENSITIVITY
+	AH_IEEE80211_SET_ED_THRESHOLD        /* set energy detection threshold */
+	AH_IEEE80211_SET_RXDESENS            /* set rx sensitivity */
+//#endif
+//#ifdef AH_SUPPORT_PER_VLAN_GTK
+	AH_IEEE80211_GET_ALL_STA_VLAN_GTK
+//#endif
+	AH_IEEE80211_SET_RRM_NBR_LIST_DUAL_BAND
+	AH_IEEE80211_SET_RRM_NBR_LIST_BAND_MAX
+	AH_IEEE80211_SET_BSS_TRANS_DISASSOC_IMMT
+	AH_IEEE80211_SET_BSS_TRANS_DISASSOC_IMMT_TIMER
+	AH_IEEE80211_SET_11KV_FAILED_ACTIONS
+//#ifdef AH_NETWORK360_CLT_CAPS
+	AH_IEEE80211_CLT_CAPS_DEL_STA
+//#endif
+//#ifdef AH_NETWORK360_WIFI_STATS
+	AH_IEEE80211_SQ_GROUP_RANGE
+//#endif
+//#ifdef AH_SUPPORT_11AX
+	AH_IEEE80211_SET_HE_MCS_MAP       /* set he mcs map */
+	AH_IEEE80211_GET_HE_MCS_MAP       /* get he mcs map */
+	AH_IEEE80211_SET_HE_OFDMA_DL		/* configure downlink OFDMA */
+	AH_IEEE80211_GET_HE_OFDMA_DL		/* get the status of downlink OFDMA */
+//#ifdef AH_SUPPORT_11AX_FEATURES
+	AH_IEEE80211_SET_HE_OFDMA_UL		/* configure uplink OFDMA */
+	AH_IEEE80211_GET_HE_OFDMA_UL		/* get the status of uplink OFDMA */
+	AH_IEEE80211_SET_HE_BSS_COLOR		/* configure BSS color */
+	AH_IEEE80211_GET_HE_BSS_COLOR		/* get the status of BSS color */
+	AH_IEEE80211_SET_TWT		        /* configure TWT */
+	AH_IEEE80211_GET_TWT		        /* get the status of TWT */
+//#endif
+//#endif
+	AH_IEEE80211_GET_CHNL_STATS     /* get channel util, tx util and rx util etc */
+	AH_IEEE80211_GET_TXCORE         /* get txcore of radio */
+	AH_IEEE80211_GET_HW_PWR_LIMIT   /* get hw power limit of radio */
+	AH_IEEE80211_GET_DFS_STATE      /* get DFS state of radio */
+//#ifdef AH_SUPPORT_ADSP_SENSOR
+	AH_IEEE80211_SET_ADSP_SENSOR_CFG
+	AH_IEEE80211_GET_ADSP_SENSOR_CFG
+	AH_IEEE80211_SET_WIPSK_CHANSPEC
+//#endif
+	AH_IEEE80211_SEND_CSA		/* send channel switch announcement */
+	AH_IEEE80211_GET_IF_COUNTRY           /* get interface brcm cc */
+	AH_IEEE80211_SET_RSNXE_CAP		/* set RSN Extended IE capabilities*/
+//#ifdef AH_SUPPORT_AFC
+	AH_IEEE80211_GET_AFC_INFO	/* get AFC info */
+//#endif
+//#ifdef AH_SUPPORT_11MC
+    AH_IEEE80211_SET_FTM           /* set FTM and LCI */
+//#endif
+//#if AH_SUPPORT_USEG
+	AH_IEEE80211_SET_USEG_BCAST_FILTER   /* set PCG broadcast filter */
+	AH_IEEE80211_SET_USEG_MCAST_FILTER   /* set PCG multicast filter */
+//#endif
+)
+
+func Ah_ioctl(fd uintptr, op, argp uintptr) error {
+	        _, _, errno := syscall.RawSyscall(syscall.SYS_IOCTL, uintptr(fd), uintptr(op), argp)
+        if errno != 0 {
+                return errno
+        }
+        return nil
+}
+
+func Ah_ifname_radio2vap(radio_name string ) string {
+    switch {
+    case radio_name == "wifi0":
+        return "wifi0.1"
+    case radio_name == "wifi1":
+        return "wifi1.1"
+    case radio_name == "wifi2":
+        return "wifi2.1"
+    default:
+        return "invalid"
+    }
+}
+
+/*
+ * Convert MHz frequency to IEEE channel number.
+ */
+func FreqToChan(freq uint16) uint16 {
```

Comment:
This function looks identical to what you have in ah_wireless.go and ah_wireless_v2.go. Consider putting it in common.